libs/miranum-core/src/lib/miranum-core.ts

```diff
@@ -53,39 +53,33 @@ export class MiranumCore {
         return generatedFiles;
     }
 
-    public async generateArtifact(artifactName: string, type: string, project: string): Promise<Artifact> {
-        return this.initArtifact(artifactName, type, project, this.getPathFromConfig(type));
+    public async generateArtifact(artifactName: string, type: string, projectName: string, projectPath: string): Promise<Artifact> {
+        /** checks if we have a projectConfig, whether we are on top-level of the project, and if so, if we have a type which has a subfolder
+         *  if so it initialises an artifact with the "pathInProject" set as the projectConfig.workspace defines it,
+         *  otherwise it initialises it as a standalone artifact
+         */
+        const lastFolder = projectPath.substring(projectPath.lastIndexOf("/")+1);
+        if(this.projectConfig && lastFolder == projectName && (type == "form" || type == "config" || type == "element-template")) {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            return this.initArtifact(artifactName, type, projectName, this.projectConfig.workspace[`${type}s`]);
+        }
+        return this.initArtifact(artifactName, type, projectName, "");
```

Comment:
```suggestion
        let pathInProject = undefined;
        if(this.projectConfig || lastFolder == projectName)) {
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            // @ts-ignore
            pathInProject = this.projectConfig.workspace[`${type}s`];
        }
        return this.initArtifact(artifactName, type, projectName, pathInProject ?? "");
```

Reply:
|| sollte definitiv && sein, denn nur wenn beides gilt soll der path ausgelesen werden
und man könnte auch einfach pathInProject als "" initialisieren, dann hat man unten kein if mehr

aber warum soll der type check weg? is der nicht auch notwendig um bpmn / dmn und alle anderen artifacts die automatisch auf toplevel sollen auszuschließen?

Reply:
Die type Unterscheidung musst du glaube ich nicht zwingend machen. Wenn z.B. der type == bpmn ist dann gibt dir `this.projectConfig.workspace[`${type}s`]` ein undefined zurück.
Wenn wir für jeden type der zutrifft einen check einbauen müssen wir am Ende das if statement jedes mal erweitern, wenn wir einen neuen Artifact type hinzufügen.

Reply:
Stimmt `this.projectConfig && lastFolder == projectName` macht mehr Sinn.